File: nc-scripts/src/app/dto/workflows/workflowstructure.ts
import { IndexTransition } from './indexTransition';
import { NodeData } from './nodeData';
import { WorkflowData } from './workflowdata';

/**
 * complete workflow definition
 */
export interface WorkflowStructure extends WorkflowData {

    /**
     * nodes in workflow
     */
    nodes: NodeData[],

    /**
     * node transitions
     */
    transitions: IndexTransition[]
}

File: ScriptService/Dto/Workflows/WorkflowStructure.cs
namespace ScriptService.Dto.Workflows {

    /// <summary>
    /// arguments used to create a new workflow
    /// </summary>
    public class WorkflowStructure : WorkflowData {
        
        /// <summary>
        /// nodes to create
        /// </summary>
        public NodeData[] Nodes { get; set; }

        /// <summary>
        /// transitions to create
        /// </summary>
        public IndexTransition[] Transitions { get; set; }
    }
}